### PR TITLE
register feature

### PR DESCRIPTION
**Pull request by MohabElziny**

### CODE REVIEW

File: app/src/main/java/com/example/a7asryan/register/view/Register.kt [hunk 1]
```diff
@@ -35,122 +34,40 @@ class Register : Fragment() {
...
-                        setUserPOJO(
-                            userName,
-                            email,
-                            password,
-                            phoneNumber
-                        )
-                    )
-                } else {
-                    binding.textErrorMessage.visibility = View.VISIBLE
+            val userName = binding.userName.editableText.toString().trim()
```
Comment (RemonShehata) [inline app/src/main/java/com/example/a7asryan/register/view/Register.kt]: `editableText.toString().trim` this part is repeated. any ideas?

File: app/src/main/java/com/example/a7asryan/register/viewmodel/RegisterViewModel.kt [hunk 2]
```diff
@@ -9,20 +11,89 @@ import kotlinx.coroutines.launch
...
+            } else {
+                insertUser(user)
+                _registerResult.value =RegisterResult.RegisterSuccessful
+            }
         }
     }
 
-    fun insertUser(user: User) {
+    private fun insertUser(user: User) {
         viewModelScope.launch {
```
Comment (RemonShehata) [inline app/src/main/java/com/example/a7asryan/register/viewmodel/RegisterViewModel.kt]: This can be `suspend` , you are launching twice.

**Review by RemonShehata — APPROVED** [review]

File: app/src/main/java/com/example/a7asryan/register/view/Register.kt (hunk 1 above)
Comment (MohabElziny) [reply app/src/main/java/com/example/a7asryan/register/view/Register.kt]: we can use the extension function for TextInputEditText provide me this part?
for example:
```
fun TextInputEditText.getTrimmedText(): String {
        return this.editableText.toString().trim()
    }
```

File: app/src/main/java/com/example/a7asryan/register/viewmodel/RegisterViewModel.kt (hunk 2 above)
Comment (MohabElziny) [reply app/src/main/java/com/example/a7asryan/register/viewmodel/RegisterViewModel.kt]: You are right ^^

File: app/src/main/java/com/example/a7asryan/register/view/Register.kt (hunk 1 above)
Comment (RemonShehata) [reply app/src/main/java/com/example/a7asryan/register/view/Register.kt]: bravo

**Review by RemonShehata — APPROVED** [review]

File: app/src/main/java/com/example/a7asryan/register/view/Register.kt [hunk 3]
```diff
@@ -35,122 +35,44 @@ class Register : Fragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        var userName: String = ""
-        var email: String = ""
-        var password: String = ""
-        var phoneNumber: String = ""
         binding.btnRegister.setOnClickListener {
```
Comment (AhmedAbdallah125) [inline app/src/main/java/com/example/a7asryan/register/view/Register.kt:40]: make here visibility is invisible

Comment (MohabElziny) [reply app/src/main/java/com/example/a7asryan/register/view/Register.kt:40]: what is this?

Comment (AhmedAbdallah125) [reply app/src/main/java/com/example/a7asryan/register/view/Register.kt:40]: i mean visibility of error text

**Pull request merged**